helm-charts/splunk-otel-collector/templates/daemonset.yaml

```
@@ -74,9 +74,26 @@ spec:
       affinity:
         {{ toYaml . | nindent 8 }}
       {{- end }}
-      {{- with .Values.tolerations }}
       tolerations:
-        {{ toYaml . | nindent 8 }}
+      {{- if and (.Values.tolerations) (ne (len .Values.tolerations) 0) }}
+        {{ toYaml .Values.tolerations | nindent 8 }}
+      {{- else }}
+        - key: node-role.kubernetes.io/master
+          effect: NoSchedule
+          operator: Equal
```

```suggestion
          operator: Exists
```

From the docs 
```
The default value for operator is Equal.

A toleration "matches" a taint if the keys are the same and the effects are the same, and:
the operator is Exists (in which case no value should be specified), or
the operator is Equal and the values should be equal.
```
Since we are not specifying a value to match against, `Exists` seems to be the correct operator type here. Weird though that this has been working w/o the value.

---

For deployed nodes the standard value for these default tolerations is an empty string which is why the default `Equal` operator works. I have no strong opinion here to prefer one of the `Equal` and `Exists` operators here, I'll add the requested changes.